pkg/operator/ceph/object/zone/dependents.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2022 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package zone
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	v1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/daemon/ceph/client"
+	"github.com/rook/rook/pkg/operator/ceph/object"
+	"github.com/rook/rook/pkg/util/dependents"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func CephObjectZoneDependents(
+	clusterdCtx *clusterd.Context,
+	clusterInfo *client.ClusterInfo,
+	zone *v1.CephObjectZone,
+	objContext *object.Context,
+) (*dependents.DependentList, error) {
+	nsName := fmt.Sprintf("%s/%s", zone.Namespace, zone.Name)
+	baseErrMsg := fmt.Sprintf("failed to get dependents of CephObjectZone %q", nsName)
+
+	deps := dependents.NewDependentList()
+	zoneIsMaster, err := object.CheckZoneIsMaster(objContext)
+	if err != nil {
+		return deps, errors.Wrapf(err, "failed to determine if zone %q is master", objContext.Zone)
+	}
+	if zoneIsMaster {
+		realmArg := fmt.Sprintf("--rgw-realm=%s", objContext.Realm)
+		zoneGroupArg := fmt.Sprintf("--rgw-zonegroup=%s", objContext.ZoneGroup)
+		zoneGroupOutput, err := object.RunAdminCommandNoMultisite(objContext, true, "zonegroup", "get", realmArg, zoneGroupArg)
+		if err != nil {
+			return deps, errors.Wrapf(err, "failed to fetch zonegroup %q", objContext.ZoneGroup)
+		}
+		zoneGroupJson, err := object.DecodeZoneGroupConfig(zoneGroupOutput)
+		if err != nil {
+			return deps, errors.Wrapf(err, "failed to decode zonegroup %q", objContext.ZoneGroup)
+		}
+		// if only master zone remains then it is okay delete
+		if len(zoneGroupJson.Zones) > 1 {
+			deps.Add("Zone is master", zone.Name)
+		}
+	}
+
+	// CephObjectStores
+	stores, err := clusterdCtx.RookClientset.CephV1().CephObjectStores(zone.Namespace).List(clusterInfo.Context, metav1.ListOptions{})
```

Comment:
It's best to use the ListOptions to filter all CephObjectStores that set their zone to the name of this zone. That way, all returned items are sure to reference this zone.

Reply:
@BlaineEXE : field selector is not working for `cephobjectstore`
```
# kubectl -n rook-ceph get cephobjectstores.ceph.rook.io --field-selector Spec.Zone.Name=new-zone
Error from server (BadRequest): Unable to find "ceph.rook.io/v1, Resource=cephobjectstores" that match label selector "", field selector "Spec.Zone.Name=new-zone": field label not supported: Spec.Zone.Name
```
tried with `spec.Zone.Name` and `spec.zone.name` as well. Also used different field such as `Status.Phase` returned similar error